### DISCUSSION TITLE
zone2sql says: Warning! Skipping '' zone 'mydomain.com'

### DISCUSSION THREAD

**Discussion by webhype** [PowerDNS/pdns#13496] [Q&A]
I am trying to migrate a BIND-style zone file to SQLITE3 using `zone2sql`.
```
$ systemctl stop pdns # make sure only one process tries to write to sqlite3
$ zone2sql --named-conf=/etc/powerdns/named.conf --gsqlite --verbose | sqlite3 /var/lib/powerdns/pdns.sqlite3
```
Expected result: That the domain be added to the sqlite3 db.

In-fact result: 
```
Domain mydomain.com lives in file '/etc/powerdns/zones/mydomain.com.zone'
 Warning! Skipping '' zone 'mydomain.com'
100% done
0 domains were fully parsed, containing 0 records
```
Contents of etc/powerdns/zones/mydomain.com.zone:
```
$TTL 86400
@   IN SOA ns1.mydomain.com. hostmaster.mydomain.com. (
                2022040101      ; serial number
                3600            ; refresh period
                1800            ; retry interval
                604800          ; expire time
                86400   ; minimum TTL
)

                IN NS ns1.mydomain.com.
                IN NS ns2.mydomain.com.

ns1             IN A  123.45.67.89
ns2             IN A  123.45.67.90
_acme-challenge TXT     "SOME_CHALLENGE"
@               IN A  123.45.67.89
*               IN A  123.45.67.89
www             IN A  123.45.67.89
```
Hitting a wall here. Neither zone2sql, not PDNS log shows any errors, no peep, nothing. I get the same result whether PDNS is running while I run the zone2sql file or not (I stopped it to exclude the possibility of the sqlite3 db being read-only). But PDNS starts fine and if there's a problem with write access, it reports that immediately. What could be wrong?

Really bad form that the zone2sql tool gives such cryptic non-error messages. It's binary so difficult to guess what it actually does.

**Comment by Habbie** [comment]
> Really bad form

This is not the complaints department.

> It's binary so difficult to guess what it actually does.

https://github.com/PowerDNS/pdns/blob/master/pdns/zone2sql.cc

> Warning! Skipping '' zone 'mydomain.com'

This means the domain did not have a `type` in your `named.conf`. You should give it one. `primary` is a good value.
